src/main/java/com/esri/samples/editing/edit_and_sync_features/EditAndSyncFeaturesSample.java

```diff
@@ -0,0 +1,403 @@
+      controlsVBox.setPadding(new Insets(10.0));
+      controlsVBox.setMaxSize(180, 20);
+      controlsVBox.getStyleClass().add("panel-region");
+
+      // create button for user interaction
+      geodatabaseButton = new Button("Generate Geodatabase");
+      geodatabaseButton.setMaxWidth(Double.MAX_VALUE);
+      controlsVBox.getChildren().add(geodatabaseButton);
+
+      // create progress indicator
+      progressIndicator = new ProgressIndicator();
+      progressIndicator.setVisible(false);
+
+      // add listener to handle generate/sync geodatabase button
+      geodatabaseButton.setOnAction(e -> {
+        if (currentEditState == EditState.NOTREADY) {
+          // update button text and disable button
+          geodatabaseButton.setDisable(true);
+          geodatabaseButton.setText("Generating Geodatabase...");
+
+          // show the extent of the geodatabase to be generated
+          final SimpleLineSymbol boundarySymbol = new SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, 0xFF0000FF, 5);
+          final Envelope extent = mapView.getVisibleArea().getExtent();
+          Graphic boundary = new Graphic(extent, boundarySymbol);
+          graphicsOverlay.getGraphics().add(boundary);
+
+          // generate Geodatabase for the chosen extent
+          generateGeodatabase(extent);
+        } else if (currentEditState == EditState.READY) {
+          syncGeodatabase();
+        }
+      });
+
+      // handle clicks on the map view to select and move features
+      mapView.setOnMouseClicked((event) -> {
+        if (event.isStillSincePress() && event.getButton() == MouseButton.PRIMARY) {
+          // get screen point where user clicked
+          Point2D screenPoint = new Point2D(event.getX(), event.getY());
+
+          // get map location corresponding to screen point
+          Point mapPoint = mapView.screenToLocation(screenPoint);
+
+          // iterate through all feature layers in the map
+          for (Layer layer : mapView.getMap().getOperationalLayers()) {
+            final FeatureLayer featureLayer = (FeatureLayer) layer;
+
+            // identify any clicked feature
+            ListenableFuture<IdentifyLayerResult> results = mapView.identifyLayerAsync(featureLayer, screenPoint, 1, false, 1);
+            results.addDoneListener(() -> {
+              try {
+                // get selected feature, if it is a point
+                List<GeoElement> elementList = results.get().getElements();
+                if (elementList.size() > 0 && elementList.get(0) instanceof ArcGISFeature && elementList.get(0).getGeometry().getGeometryType() == GeometryType.POINT) {
+
+                  // clicked on a feature, select it
+                  ArcGISFeature selectedFeature = (ArcGISFeature) elementList.get(0);
+                  // TODO: this doesn't clear the selection in all FL, hence some elements stay selected?
```

Comment:
all TODOs should be cleared before merging to master: or is this a bug you are highlighting?